doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,785 @@
+Before=xroad-confclient.service
+Before=xroad-jetty.service
+[Service]
+User=xroad
+Group=xroad
+Type=oneshot
+Environment=XROAD_USER={{ xroad_slave_ssh_user }}
+Environment=MASTER={{ master_host }}
+ExecStart=/usr/bin/rsync -e "ssh -o ConnectTimeout=5 " -aqz --timeout=10 --delete-delay --exclude db.properties --exclude "/conf.d/node.ini" --exclude "*.tmp" --exclude "/postgresql" --exclude "/nginx" --exclude "/globalconf" --delay-updates --log-file=/var/log/xroad/slave-sync.log ${XROAD_USER}@${MASTER}:/etc/xroad/ /etc/xroad/
+[Install]
+WantedBy=multi-user.target
+WantedBy=xroad-proxy.service
+```
+
+Add a timer for periodic updates.
+
+Create a new file `/etc/systemd/xroad-sync.timer`:
+
+```
+[Unit]
+Description=Sync X-Road configuration
+[Timer]
+OnBootSec=60
+OnUnitActiveSec=60
+[Install]
+WantedBy=timers.target
+```
+
+Configure SELinux to allow `rsync` to be run as a `systemd` service
+
+```
+setsebool -P rsync_client 1
+setsebool -P rsync_full_access 1
+```
+Enable the services:
+```
+systemctl enable xroad-sync.timer xroad-sync.service
+systemctl start xroad-sync.timer
+```
+
+#### 5.2.2 Ubuntu: Use upstart and cron for configuration synchronization
+
+Create the main upstart task for syncing.
+
+Create a new file: `/etc/init/xroad-sync.conf`:
+```
+# xroad-sync
+env XROAD_USER={{ xroad_slave_ssh_user }}
+env MASTER={{ master_host }}
+task
+script
+  su -s /bin/sh -c 'exec "$0" "$@"' xroad -- rsync -e "ssh -o ConnectTimeout=5 " -aqz --timeout=10 --delete-delay --exclude db.properties --exclude "/conf.d/node.ini" --exclude "*.tmp" --exclude "/postgresql" --exclude "/nginx" --exclude "/globalconf" --delay-updates --log-file=/var/log/xroad/slave-sync.log ${XROAD_USER}@${MASTER}:/etc/xroad/ /etc/xroad/
+end script
+```
+
+Create a helper task that ensures that the sync task is executed before the services are started.
+
+Create a new file `/etc/init/xroad-sync-wait.conf`:
+```
+# wait for xroad-sync to complete before starting services
+start on starting xroad-proxy or starting xroad-jetty or starting xroad-signer or starting xroad-confclient
+stop on stopped xroad-sync
+instance $JOB
+normal exit 2
+task
+script
+  echo "$JOB"
+  start xroad-sync || :
+  sleep 360
+end script
+```
+
+Add a cron job to periodically start the sync task as upstart does not have a timer facility.
+
+Create a new file `/etc/cron.d/xroad-state-sync`:
+```
+* * * * * root /sbin/initctl --quiet start xroad-sync
+```
+
+**A note about the `rsync` options:**
+>
+>* `--delay-updates` and `--delete-delay` make the sync more atomic by delaying modifications until data has been
+> downloaded. It is not fully atomic, however, since the files will be moved into place one by one. If the synchronization
+> is disrupted, no modifications will be made.
+> * low connect timeout (5 seconds) and receive timeout (10 seconds) ensure that the synchronization won't hang if e.g.
+> a network connection fails.
+
+
+### 5.3 Set up log rotation for the sync log on the slave nodes
```

Comment:
Added some details near the service/task creation about what the sync log is and also at the beginning. Also added the bit about being on slave nodes although the title ".. on the slave nodes" is somewhat unambiguous.